contracts/implementations/nodes/AftermarketDevice.sol

```diff
@@ -404,6 +404,53 @@ contract AftermarketDevice is
         emit AftermarketDevicePaired(aftermarketDeviceNode, vehicleNode, owner);
     }
 
+    /**
+     * @dev Unpairs an aftermarket device from a vehicles
+     * Both vehicle and AD owners can unpair.
+     * @dev Caller must have the admin role
+     * @param aftermarketDeviceNode Aftermarket device node id
+     * @param vehicleNode Vehicle node id
+     */
+    function unpairAftermarketDevice(
+        uint256 aftermarketDeviceNode,
+        uint256 vehicleNode
+    ) external {
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        address adIdProxyAddress = AftermarketDeviceStorage
+            .getStorage()
+            .idProxyAddress;
+
+        if (!INFT(adIdProxyAddress).exists(aftermarketDeviceNode))
+            revert Errors.InvalidNode(adIdProxyAddress, aftermarketDeviceNode);
+        if (!INFT(vehicleIdProxyAddress).exists(vehicleNode))
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleNode);
+        if (
+            ms.links[vehicleIdProxyAddress][vehicleNode] !=
+            aftermarketDeviceNode
+        ) revert VehicleNotPaired(vehicleNode);
+        if (ms.links[adIdProxyAddress][aftermarketDeviceNode] != vehicleNode)
+            revert AdNotPaired(aftermarketDeviceNode);
```

Comment:
If we pass the first check but fail the second, that's catastrophic, no?

Reply:
Kind of, bc those two links are always supposed to be updated together. Might be worth combine these two `ifs` then

Reply:
Well, you have the two events, let's use them. I just wanted to point it out.